go.mod

```diff
@@ -1,16 +1,18 @@
 module github.com/hashicorp/boundary-plugin-minio
 
-go 1.22.2
+go 1.23.4
```

Comment:
FYI it's potentially restrictive to include specific patch versions in the go directive of _libraries_. See https://github.com/grpc/grpc-go/pull/7624#discussion_r1757159186, https://github.com/grpc/grpc-go/pull/7831, https://github.com/golang/go/issues/65847 for some discussion.

Reply:
Thanks for calling this out, however it seems someone should have commented on one of the library's being used since trying to update this `go mod tidy` then causes:
```
go 1.23.1

toolchain go1.23.4
```
I did update the madmin package